### PR TITLE
Add metadata to combine_channels

### LINKED ISSUE #10499: mne.channels.combine_channels(epochs) drops metadata
I think `mne.channels.combine_channels(epochs)` drops metadata (i.e. there is no `epochs.metadata` attribute in the resulting `Epochs` objects). I wanted to make sure that this is not intended – if it is a bug I'm happy to submit a PR.

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #10499.

### CODE REVIEW

File: mne/channels/channels.py [hunk 1]
```diff
@@ -1710,6 +1710,8 @@ def combine_channels(inst, groups, method='mean', keep_stim=False,
     elif isinstance(inst, BaseEpochs):
         combined_inst = EpochsArray(new_data, info, events=inst.events,
                                     tmin=inst.times[0])
+        if inst.metadata is not None:
+            combined_inst.metadata = inst.metadata
```
Comment (agramfort) [inline mne/channels/channels.py:1714]:
```suggestion
            combined_inst.metadata = inst.metadata.copy()
```
I think it's safer.

can you also add a what's new entry? and update a test 🙏

Comment (cbrnr) [reply mne/channels/channels.py:1714]: Yes, will do!

**Review by agramfort — APPROVED** [review]

**Comment by cbrnr** [comment]
@larsoner feel free to merge (or let me know if you want me to change anything).

**Comment by larsoner** [comment]
Thanks @cbrnr !

**Comment by cbrnr** [comment]
Could we backport this fix @larsoner?

**Comment by agramfort** [comment]
done

**Pull request merged**